code/obj/machinery/pipe/pipe_dispenser.dm

```diff
@@ -11,62 +21,87 @@ TYPEINFO(/obj/machinery/disposal_pipedispenser)
 	anchored = 1
 	deconstruct_flags = DECON_SCREWDRIVER | DECON_WRENCH | DECON_CROWBAR | DECON_WELDER | DECON_WIRECUTTERS
 
-/obj/machinery/disposal_pipedispenser/attack_hand(mob/user)
-	if(..())
-		return
-
-	var/dat = {"<b>Disposal Pipes</b><br><br>
-<A href='?src=\ref[src];dmake=0'>Pipe</A><BR>
-<A href='?src=\ref[src];dmake=1'>Bent Pipe</A><BR>
-<A href='?src=\ref[src];dmake=2'>Junction</A><BR>
-<A href='?src=\ref[src];dmake=3'>Y-Junction</A><BR>
-<A href='?src=\ref[src];dmake=4'>Trunk</A><BR>
-"}
-
-	user.Browse("<HEAD><TITLE>Disposal Pipe Dispenser</TITLE></HEAD><TT>[dat]</TT>", "window=pipedispenser")
-	return
-
-// 0=straight, 1=bent, 2=junction-j1, 3=junction-j2, 4=junction-y, 5=trunk
-
-
-/obj/machinery/disposal_pipedispenser/Topic(href, href_list)
-	if(..())
+	var/dispenser_ready = TRUE
+	var/dispenser_delay = 5 DECI SECONDS
+	var/mobile = FALSE
+	var/datum/action/bar/icon/timer
+
+/obj/machinery/disposal_pipedispenser/ui_interact(mob/user, datum/tgui/ui)
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "PipeDispenser")
+		ui.open()
+
+/obj/machinery/disposal_pipedispenser/ui_data(mob/user)
+	. = list(
+		"dispenser_ready" = src.dispenser_ready,
+		"windowName" = src.name,
```

Comment:
Minor, but I would expect this to be static. No less static than `mobile`, anyway.

Reply:
Done.